blog/2018-01-09-smart-testing-for-fast-feedback.textile

```diff
@@ -0,0 +1,57 @@
+
+For almost every new feature added to the application, there is also a test written. This way, whenever someone changes the user interface, the system will be able to catch the differences and signal the team about any issues. They're still not integrated into the main pipeline. The plan is to merge them in to gain more confidence in what they build. The tests should run whenever anyone pushes changes to the git repository.
+
+Work starts on the new CD pipeline and everything is set up accordingly. The pipeline structure looks like below:
+
+code quality → unit tests → build → staging deployment → e2e test
+
+For every developer pushing code to the repo, the system kicks off a process and runs the whole pipeline.
+
+Running it at every code push is costly, not only in terms of resource utilization (machines, networks, servers) but also in terms of developer time. Whenever the build breaks, all work should stop and the build should be fixed before introducing more features (and possibly more problems). The developer has to wait for the build to pass in order to move on.
+
+Imagine there are 2 or more developers working on that branch. They would have to push to version control only when they are sure the functionality is 100% ready. Otherwise, we would break the application for every person working on the project. This leads to more problems.
+
+Putting it all together it is very important to know that we should refrain from optimizing for the best case. It is always better to optimize for failure, accept it, embrace it.
+
+For teams doing some form of Continuous Integration/Delivery/Deployment, if the pipeline is slow, then the feedback loop is slow. If the code takes 30 minutes to pass through the pipeline and build only to fail in UAT or even worse, production, precious time and money are wasted.
+
+Feedback is very important but if it comes slow, or after you pushed to production, it has no preventative value. It only serves as a lesson learned. Having feedback from the early beginnings of a project, and receiving it fast, is the cornerstone to Continuous Delivery / Deployment. It allows you to optimize ahead of time, with little to no impact on your users.
+
+Hence, going fast is not only about shipping fast but also about failing fast. The faster you fail, the cheaper the solution.
+
+h2. How does Smart Testing shorten the Feedback Loop?
+
+There are a couple things you can do to shorten the feedback loop. You should absolutely invest in your tests and make them faster. Besides trying to implement faster tests, however, you can also optimize the way you run them.
```

Comment:
I would keep only the first sentence:
`There are a couple things you can do to shorten the feedback loop.` then name the possibilities:
* don't run the tests
* make test faster
* parallelization
* buy more resources
* prioritize
* minimize test suite

then describe the problems of the approaches - not running the tests is not a good idea; making test faster is hard (as you wrote there); parallelization is great, but you can do it to some limit; buying more resources - the limit is the budget. So we have prioritization and minimization of the test suite, but how to do it? Which tests should we choose? and then you can jump to the options/strategies etc...